test/ziggurat/messaging/producer_test.clj

```diff
@@ -51,7 +51,73 @@
           (let [message-from-mq (rmq/get-message-from-channel-delay-queue topic-entity channel)]
             (producer/retry-for-channel message-from-mq channel)))
         (let [message-from-mq (rmq/get-msg-from-channel-dead-queue topic-entity channel)]
-          (is (= expected-message-payload message-from-mq)))))))
+          (is (= expected-message-payload message-from-mq))))))
+
+  (testing "message in channel will be retried with linear queue timeout"
+    (with-redefs [ziggurat-config (constantly (assoc (ziggurat-config)
+                                                     :stream-router {:default {:channels {:linear-retry {:retry {:count 5
+                                                                                                                 :enabled true
+                                                                                                                 :queue-timeout-ms 2000}}}}}))]
+      (fix/with-queues
+        {:default {:handler-fn #(constantly nil)
+                   :linear-retry #(constantly nil)}}
+        (let [retry-count (atom 2)
+              topic-entity :default
+              channel :linear-retry
+              message-payload {:message {:foo "bar"}  :topic-entity topic-entity :retry-count @retry-count}
+              expected-message-payload (assoc message-payload :retry-count 0)]
+          (producer/retry-for-channel message-payload channel)
+          (while (> @retry-count 0)
+            (swap! retry-count dec)
+            (let [message-from-mq (rmq/get-message-from-channel-delay-queue topic-entity channel)]
+              (producer/retry-for-channel message-from-mq channel)))
+          (let [message-from-mq (rmq/get-msg-from-channel-dead-queue topic-entity channel)]
+            (is (= expected-message-payload message-from-mq)))))))
+
+  (testing "message in channel will be retried with exponential queue timeout"
```

Comment:
can be corrected to ".. exponential timeout calculated fro channel specific queue-timeout-ms value".

Reply:
this is updated